### ISSUE TITLE
405 Error with get_identifier_authorization

### ISSUE THREAD

**Issue by moritz89** [komuw/sewer#128]
## Which version of python are you using?
Python 3.7.5

## What operating system and version of operating system are you using?
Ubuntu 19.10

## What version of sewer are you using?
0.7.6

## What did you do? (be as detailed as you can)
Call `sewer --endpoint staging --dns cloudflare --domain "mydomain.com" --action run`

## What did you expect to see/happen/not happen?
Register a new TLS certificate

## What did you actually see/happen? 
An error during the _get_identifier_authorization_ step:

`Error: Unable to issue certificate. error=Error getting identifier authorization: status_code=405 response={'type': 'urn:ietf:params:acme:error:malformed', 'detail': 'Method not allowed', 'status': 405}`

## If you can, run sewer with loglevel set to debug; eg `sewer --loglevel DEBUG`                                           
```
chosen_dns_provider. Using cloudflare as dns provider.
get_acme_endpoints
get_acme_endpoints
Starting new HTTPS connection (1): acme-staging-v02.api.letsencrypt.org:443
https://acme-staging-v02.api.letsencrypt.org:443 "GET /directory HTTP/1.1" 200 724
get_acme_endpoints_response. status_code=200
get_acme_endpoints_response. status_code=200
create_certificate_key
create_certificate_key
create_csr
create_csr
create_account_key
create_account_key
intialise_success, sewer_version=0.7.6, domain_names=['mydomain.com'], acme_server=https://acme-staging...
intialise_success, sewer_version=0.7.6, domain_names=['mydomain.com'], acme_server=https://acme-staging...
account key succesfully written to /home/.../mydomain.com.account.key.
get_certificate
get_certificate
acme_register
acme_register
make_signed_acme_request
make_signed_acme_request
get_acme_header
get_acme_header
get_nonce
get_nonce
Starting new HTTPS connection (1): acme-staging-v02.api.letsencrypt.org:443
https://acme-staging-v02.api.letsencrypt.org:443 "GET /acme/new-nonce HTTP/1.1" 204 0
sign_message
sign_message
Starting new HTTPS connection (1): acme-staging-v02.api.letsencrypt.org:443
https://acme-staging-v02.api.letsencrypt.org:443 "POST /acme/new-acct HTTP/1.1" 201 526
acme_register_response. status_code=201. response={'key': {'kty': 'RSA', 'n': '..........', 'e': 'AQAB'}, 'contact': [], 'initialIp': '87.133.215.69', 'createdAt': '2019-12-05T10:55:52.617378487Z', 'status': 'valid'}
acme_register_response. status_code=201. response={'key': {'kty': 'RSA', 'n': '..........', 'e': 'AQAB'}, 'contact': [], 'initialIp': '87.133.215.69', 'createdAt': '2019-12-05T10:55:52.617378487Z', 'status': 'valid'}
acme_register_success
acme_register_success
apply_for_cert_issuance
apply_for_cert_issuance
make_signed_acme_request
make_signed_acme_request
get_acme_header
get_acme_header
get_nonce
get_nonce
Starting new HTTPS connection (1): acme-staging-v02.api.letsencrypt.org:443
https://acme-staging-v02.api.letsencrypt.org:443 "GET /acme/new-nonce HTTP/1.1" 204 0
sign_message
sign_message
Starting new HTTPS connection (1): acme-staging-v02.api.letsencrypt.org:443
https://acme-staging-v02.api.letsencrypt.org:443 "POST /acme/new-order HTTP/1.1" 201 372
apply_for_cert_issuance_response. status_code=201. response={'status': 'pending', 'expires': '2019-12-12T10:55:54.147399744Z', 'identifiers': [{'type': 'dns', 'value': 'mydomain.com'}], 'authorizations': ['https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/25378193'], 'finalize': 'https://acme-staging-v02.api.letsencrypt.org/acme/finalize/11722005/64585723'}
apply_for_cert_issuance_response. status_code=201. response={'status': 'pending', 'expires': '2019-12-12T10:55:54.147399744Z', 'identifiers': [{'type': 'dns', 'value': 'mydomain.com'}], 'authorizations': ['https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/25378193'], 'finalize': 'https://acme-staging-v02.api.letsencrypt.org/acme/finalize/11722005/64585723'}
apply_for_cert_issuance_success
apply_for_cert_issuance_success
get_identifier_authorization
get_identifier_authorization
Starting new HTTPS connection (1): acme-staging-v02.api.letsencrypt.org:443
https://acme-staging-v02.api.letsencrypt.org:443 "GET /acme/authz-v3/25378193 HTTP/1.1" 405 103
get_identifier_authorization_response. status_code=405. response={'type': 'urn:ietf:params:acme:error:malformed', 'detail': 'Method not allowed', 'status': 405}
get_identifier_authorization_response. status_code=405. response={'type': 'urn:ietf:params:acme:error:malformed', 'detail': 'Method not allowed', 'status': 405}
Error: Unable to issue certificate. error=Error getting identifier authorization: status_code=405 response={'type': 'urn:ietf:params:acme:error:malformed', 'detail': 'Method not allowed', 'status': 405}
Error: Unable to issue certificate. error=Error getting identifier authorization: status_code=405 response={'type': 'urn:ietf:params:acme:error:malformed', 'detail': 'Method not allowed', 'status': 405}
Traceback (most recent call last):
  File "/home/moritz/.virtualenvs/core-2f8UfBO0/bin/sewer", line 8, in <module>
    sys.exit(main())
  File "/home/moritz/.virtualenvs/core-2f8UfBO0/lib/python3.7/site-packages/sewer/cli.py", line 332, in main
    certificate = client.cert()
  File "/home/moritz/.virtualenvs/core-2f8UfBO0/lib/python3.7/site-packages/sewer/client.py", line 749, in cert
    return self.get_certificate()
  File "/home/moritz/.virtualenvs/core-2f8UfBO0/lib/python3.7/site-packages/sewer/client.py", line 738, in get_certificate
    raise e
  File "/home/moritz/.virtualenvs/core-2f8UfBO0/lib/python3.7/site-packages/sewer/client.py", line 698, in get_certificate
    identifier_auth = self.get_identifier_authorization(url)
  File "/home/moritz/.virtualenvs/core-2f8UfBO0/lib/python3.7/site-packages/sewer/client.py", line 406, in get_identifier_authorization
    response=self.log_response(get_identifier_authorization_response),
ValueError: Error getting identifier authorization: status_code=405 response={'type': 'urn:ietf:params:acme:error:malformed', 'detail': 'Method not allowed', 'status': 405}
```

**Comment by moritz89** [comment]
Seems to have been a temporary problem. Tested on 3 different machines in different subnets and registering different domains.

**Comment by mmaney** [comment]
As mentioned in the duplicate entry, #135, this is due to the announced deprecation of unsigned GET requests for most of the ACME API.  I'd forgotten this change was coming when it came time to renew some certs here, and I wanted to test the new and improved script using staging before hitting production, which of course wasn't going to happen.

I have since hacked up client.py to the point that I can run a simple, single-host certificate fetch successfully.  Patch attached, as was doing this in a detached repo... and besides, this is still a WIP.  I would expect it to apply cleanly against 0.7.9, but I did have a new, definitely not yet ready for anyone but me, dns provider because that's the one I needed to use in the local tree before I started down the path to put-as-get.

[0160-Working-for-simple-single-host-cert-run-against-stag.txt](https://github.com/komuw/sewer/files/4137985/0160-Working-for-simple-single-host-cert-run-against-stag.txt)
